bindings/ros/aditof_roscpp/src/message_factory.cpp

```diff
@@ -0,0 +1,13 @@
+#include "message_factory.h"
+
+AditofSensorMsg *
+MessageFactory::create(const std::shared_ptr<aditof::Camera> &camera,
+                       MessageType type) {
+    switch (type) {
+    case MessageType::sensor_msgs_PointCloud2:
+        return new PointCloud2Msg(camera);
+    default:
```

Comment:
Default might do more harm than good. I would remove it. This way if at some point in future one adds a new element to the enum but forgets to handle it in this switch, the compiler will tell us that. Otherwise the new element will fall to default and this will be probably detected at run-time, if lucky.